### PR TITLE
Don't unconditionally enable bevy_render or bevy_assets if mutli-threaded feature is enabled

### PR DESCRIPTION
**Pull request by AxiomaticSemantics**
# Objective

bevy_render has been set to be automatically enabled if mutlti-threaded feature is

## Solution

make it conditional

### CODE REVIEW

File: crates/bevy_internal/Cargo.toml [hunk 1]
```diff
@@ -77,7 +77,7 @@ serialize = [
 multi-threaded = [
   "bevy_asset/multi-threaded",
   "bevy_ecs/multi-threaded",
-  "bevy_render/multi-threaded",
+  "bevy_render?/multi-threaded",
```
Comment (alice-i-cecile) [inline crates/bevy_internal/Cargo.toml:80]: Neat, I hadn't seen this syntax before. Do you have docs to link for this?

Comment (alice-i-cecile) [reply crates/bevy_internal/Cargo.toml:80]: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html doesn't appear to mention it.

Comment (alice-i-cecile) [reply crates/bevy_internal/Cargo.toml:80]: Aha: https://doc.rust-lang.org/cargo/reference/features.html

> The "package-name/feature-name" syntax will also enable package-name if it is an optional dependency. Often this is not what you want. You can add a ? as in "package-name?/feature-name" which will only enable the given feature if something else enables the optional dependency.

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_internal/Cargo.toml [hunk 2]
```diff
@@ -77,7 +77,7 @@ serialize = [
 multi-threaded = [
   "bevy_asset/multi-threaded",
```
Comment (mockersf) [inline crates/bevy_internal/Cargo.toml:78]:
```suggestion
  "bevy_asset?/multi-threaded",
```
could you also change it for bevy_asset?

Comment (AxiomaticSemantics) [reply crates/bevy_internal/Cargo.toml]: Done

**Review by james7132 — APPROVED** [review]

**Pull request merged**